sklearn/model_selection/tests/test_search.py

```diff
@@ -943,7 +943,10 @@ def test_search_default_iid(SearchCV, specialized_params):
     # this leads to perfect classification on one fold and a score of 1/3 on
     # the other
     # create "cv" for splits
-    cv = [[mask, ~mask], [~mask, mask]]
+    first_fold_indices = np.flatnonzero(mask)
```

Comment:
this seems to show that there was boolean support in the splitters and now there's not. I don't see how that aligns with what @glemaitre and @jnothman say?

Reply:
Maybe I was mistaken and there was already Boolean mask support in splitters. We certainly shouldn't change that if it was already supported.

I had not been trying to say that I had researched all the places in the code that this change was going to affect; rather to say that this could have effects. I thought I'd checked that there was no handling of masks between split and safe_indexin, but if they work, so be it.

Reply:
I also haven't checked extensively, but this change here in the tests suggests that there was support.

Reply:
OK so looking at the code that we had in the codebase, the `safe_indexing` was supporting boolean array-like and integer slice as well. I am adding back those supports.